### DISCUSSION TITLE
Is JetstreamContext thread safe?

### DISCUSSION THREAD

**Discussion by robertmircea** [nats-io/nats.net.v1#600] [Q&A] [answered]
Is Jetstream context (implementation of IJetstream) safe to share among many threads? I would like to know if I can register as singleton a Jetstream context with the DI container. Currently I inject a single, DI registered, NatsConnection, but I could inject actually a single IJetstream instead.

Looking at the actual `JetStream` class code, it looks it is but I would like to know for sure if a singleton is the recommended pattern of usage.

**Comment by scottf** [comment, marked as answer]
The context is essentially a functional wrapper around the connection, which is thread-safe.
